geo/src/main/java/org/odk/collect/geo/SelectionMapFragment.kt

```diff
@@ -0,0 +1,416 @@
+package org.odk.collect.geo
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.activity.OnBackPressedCallback
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import org.odk.collect.androidshared.ui.ToastUtils
+import org.odk.collect.geo.databinding.SelectionMapLayoutBinding
+import org.odk.collect.geo.databinding.SelectionSummarySheetLayoutBinding
+import org.odk.collect.geo.maps.MapFragment
+import org.odk.collect.geo.maps.MapFragment.ReadyListener
+import org.odk.collect.geo.maps.MapFragmentFactory
+import org.odk.collect.geo.maps.MapPoint
+import org.odk.collect.permissions.PermissionsProvider
+import javax.inject.Inject
+
+class SelectionMapFragment : Fragment() {
```

Comment:
Now as it is a fragment we should clear binding `onDestroyView()` https://developer.android.com/topic/libraries/view-binding#fragments

Reply:
Huh interesting. My thinking was that there would be no need for this as recreation would simply replace the reference. There's a suggestion in [this SO](https://stackoverflow.com/questions/65295104/android-view-binding-clear-binding-in-fragment-lifecycle) that one potential reason to `null` out the reference is to prevent leaks if the `Fragment` hangs around after the view is destroyed. However, it doesn't seem like that's a case anyone has a tangible example of (and [this linked article](https://www.techyourchance.com/android-fragment-lifecycle/) even refers to the practice as "premature optimization". But of course, the official docs do state that: 

> Fragments outlive their views. Make sure you clean up any references to the binding class instance in the fragment's [onDestroyView()](https://developer.android.com/reference/kotlin/androidx/fragment/app/Fragment#ondestroyview) method.

It honestly feels like View Binding wasn't though out for Fragments properly (like everything else), and they've had to add that in due to memory problems people have run into. It's not even in their [code samples](https://github.com/android/architecture-components-samples/blob/main/GithubBrowserSample/app/src/main/java/com/android/example/github/ui/user/UserFragment.kt) for instance.

After getting grumpy about all this and looking through Android's issue tracker I found [this discussion](https://issuetracker.google.com/issues/152333892) that links [this article](https://issuetracker.google.com/issues/152333892) that demonstrates using`<binding class>.bind(view)` in `onViewCreated` to avoid create binding fields. I'm going to file an issue around that being added to the View Binding docs (https://developer.android.com/topic/libraries/view-binding) and go with that to avoid needing an `onDestroyView`.